plugin/clouddns/setup.go

```diff
@@ -0,0 +1,150 @@
+package clouddns
+
+import (
+	"context"
+	"io/ioutil"
+	"strings"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/fall"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+
+	"github.com/mholt/caddy"
+	"golang.org/x/oauth2"
+	gauth "golang.org/x/oauth2/google"
+
+	gdns "google.golang.org/api/dns/v1"
+)
+
+var log = clog.NewWithPlugin("clouddns")
+
+func init() {
+	caddy.RegisterPlugin("clouddns", caddy.Plugin{
+		ServerType: "dns",
+		Action: func(c *caddy.Controller) error {
+			f := func(credential *gauth.Credentials) {
+			}
+			return setup(c, f)
+		},
+	})
+}
+
+func setup(c *caddy.Controller, f func(*gauth.Credentials)) error {
+	keys := map[string][]string{}
+	var credsFilePath string
+	// Route53 plugin attempts to find AWS credentials by using ChainCredentials.
+	// And the order of that provider chain is as follows:
+	// Static AWS keys -> Environment Variables -> Credentials file -> IAM role
+	// With that said, even though a user doesn't define any credentials in
+	// Corefile, we should still attempt to read the default credentials file,
+	// ~/.aws/credentials with the default profile.
+
+	// sharedProvider := &credentials.SharedCredentialsProvider{}
+	// var providers []credentials.Provider
+	var fall fall.F
+
+	up, _ := upstream.New(nil)
+	for c.Next() {
+		args := c.RemainingArgs()
+
+		for i := 0; i < len(args); i++ {
```

Comment:
```go
for i, arg := range args
```

Reply:
arg is not used, which is the reason for this loop. (from the legacy code)

Reply:
I added it with omitting arg